app/services/LocationService.js

```diff
@@ -79,35 +79,39 @@ export class LocationData {
       }
 
       // Backfill the stationary points, if available
-      // The assumption is that if we see a gap in the data, the
-      // best guess is that the person was in that location for
-      // the entire time.  This makes it easier for a health authority
+      // The assumption is that if we see a gap in the data, and the 
+      // device hasn't moved significantly, then we can fill in the missing data
+      // with the current location.  This makes it easier for a health authority
       // person to have a set of locations over time, and they can manually
       // redact the time frames that aren't correct.
       if (curated.length >= 1) {
         let lastLocationArray = curated[curated.length - 1];
 
-        // Figure out the time to start the backfill from.  Default is
-        // the time of the last entry in the location set.
-        // To protect ourselves, we won't backfill more than the
-        // maxBackfillTime
-        let lastRecordedTime = lastLocationArray['time'];
-        if (unixtimeUTC - lastRecordedTime > this.maxBackfillTime) {
-          lastRecordedTime = unixtimeUTC - this.maxBackfillTime;
-        }
+        let nearby = isLocationsNearby(
+          lastLocationArray['latitude'],
+          lastLocationArray['longitude'],
+          location['latitude'],
+          location['longitude'],
+        );
+        //console.log('[INFO] nearby:', nearby);
 
-        for (
-          let newTS = lastRecordedTime + this.locationInterval;
-          newTS < unixtimeUTC - this.locationInterval;
-          newTS += this.locationInterval
-        ) {
-          let lat_lon_time = {
-            latitude: lastLocationArray['latitude'],
-            longitude: lastLocationArray['longitude'],
-            time: newTS,
-          };
-          console.log('[INFO] backfill location:', lat_lon_time);
-          curated.push(lat_lon_time);
+        // Actually do the backfill if the current point is nearby the previous
+        // point and the time is within the maximum time to backfill.
+        let lastRecordedTime = lastLocationArray['time'];
+        if (nearby && unixtimeUTC - lastRecordedTime < this.maxBackfillTime) {
```

Comment:
Hah, sorry I wasnt clear this was the `isX` I was talking about.

areLocationsNearby was fine